staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -426,3 +434,29 @@ func getEnvelopePrefixTransformer(config *apiserverconfig.KMSConfiguration, enve
 		Prefix:      []byte(prefix + config.Name + ":"),
 	}, nil
 }
+
+// parseEndpoint parses the endpoint to extract schema, host or path.
+func parseEndpoint(endpoint string) (string, error) {
+	if len(endpoint) == 0 {
+		return "", fmt.Errorf("remote KMS provider can't use empty string as endpoint")
+	}
+
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", fmt.Errorf("invalid endpoint %q for remote KMS provider, error: %v", endpoint, err)
```

Comment:
Clarify that the issue is with a configuration file config, e.g. "invalid Resources[].KMS.Endpoint %q in encryption configuration for remote KMS provider..." ?